broker/src/main/java/io/camunda/zeebe/broker/system/partitions/impl/AsyncSnapshotDirector.java

```diff
@@ -235,18 +237,18 @@ private void takeSnapshot() {
     final var transientSnapshotFuture =
         stateController.takeTransientSnapshot(lowerBoundSnapshotPosition);
     transientSnapshotFuture.onComplete(
-        (optionalTransientSnapshot, snapshotTakenError) -> {
+        (transientSnapshot, snapshotTakenError) -> {
           if (snapshotTakenError != null) {
-            LOG.error("Could not take a snapshot for {}", processorName, snapshotTakenError);
-            resetStateOnFailure();
+            if (snapshotTakenError instanceof SnapshotException.SnapshotAlreadyExistsException) {
+              LOG.debug("Did not take a snapshot. {}", snapshotTakenError.getMessage());
```

Comment:
:x: Please pass the error to the logger as well.

Reply:
Is it correct that we not reset here the state?

Reply:
It is not necessary to reset here, as nothing has happened. But to avoid confusion, I will reset it once in the outer if block.

Reply:
> Please pass the error to the logger as well.

Was not sure if we have to print the entire stack trace here, since this is an expected case. That's why I only logged the message from the error. Wdyt?

Reply:
yeah you're probably right